### PR TITLE
Adds some basic ways to earn Job XP for Chefs

### PR DESCRIPTION
**Pull request by Carbadox**
## About the PR 
Adds some basic ways to earn Job XP for Chefs. Will expand more to methods + job rewards later on hopefully

## Why's this needed? 
To help expand the Job XP system for chefs

## Changelog

```changelog
(u)Carbadox
(+)Added some basic ways to earn Job XP for Chefs.
```

### CODE REVIEW

File: code/modules/food_and_drink/cakes.dm [hunk 1]
```diff
@@ -133,6 +133,8 @@
 				frostingoverlay.alpha = 255
 				src.UpdateOverlays(frostingoverlay,tag)
 				tube.reagents.trans_to(src,25)
+			if ((user.mind.assigned_role == "Chef") && (prob(25)))
+				JOB_XP(user, "Chef", 1)
```
Comment (BatElite) [inline code/modules/food_and_drink/cakes.dm:137]: Applies for every change, but I think the `JOB_XP` macro already checks if their job matches before giving XP.

**Review by ZeWaka — CHANGES_REQUESTED** [review]
you have a review.

**Comment by Carbadox** [comment]
Its been over a month

File: code/obj/kitchenspike.dm [hunk 2]
```diff
@@ -27,6 +27,8 @@
 		G.affecting.unequip_all()
 		qdel(G.affecting)
 		qdel(G)
+		if (prob(25))
+			JOB_XP(user, "Chef", 1)
```
Comment (flrsh) [inline code/obj/kitchenspike.dm:31]:
```suggestion
		JOB_XP(user, "Chef", 2)
```
Imo.

**Review by flrsh — APPROVED** [review]

File: code/modules/food_and_drink/ingredients.dm [hunk 3]
```diff
@@ -180,6 +180,8 @@
 			new /obj/item/reagent_containers/food/snacks/deviledegg(get_turf(src))
 			new /obj/item/reagent_containers/food/snacks/deviledegg(get_turf(src))
 			qdel(src)
+		if (prob(25))
```
Comment (Kyle2143) [inline code/modules/food_and_drink/ingredients.dm:183]: For safety I'd move this conditional above the qdel(src). Probably would never cause a problem, but it just doesn't make sense to have code to be executed om an object after you have it slated for garbage collection.

**Review by Kyle2143 — COMMENTED** [review]
These seem good to me at first glance. The values are a bit low, and there is plenty more room to add things to get exp like butchering, frying, gibbing. But it's a good start, and it's better to start with lower xp rates when it's new IMO.

Just fix that one line, and as flourish suggested, it would probably be fine to increase the values a bit more or remove probs if they are on things that aren't too repeatable. But idk enough about the chef role and these objects to say which they are.

**Review by Kyle2143 — APPROVED** [review]

File: code/modules/food_and_drink/ingredients.dm [hunk 4]
```diff
@@ -518,11 +536,15 @@
...
+			JOB_XP(user, "Chef", 1)
 		else if (istype(W, /obj/item/axe) || istype(W, /obj/item/circular_saw) || istype(W, /obj/item/kitchen/utensil/knife) || istype(W, /obj/item/scalpel) || istype(W, /obj/item/sword) || istype(W,/obj/item/saw) || istype(W,/obj/item/knife/butcher))
 			boutput(user, "<span class='notice'>You cut [src] into smaller pieces...</span>")
 			for(var/i = 1, i <= 3, i++)
 				new /obj/item/reagent_containers/food/snacks/ingredient/tortilla(get_turf(src))
 			qdel(src)
+		if (prob(25))
+			JOB_XP(user, "Chef", 1)
```
Comment (ZeWaka) [inline code/modules/food_and_drink/ingredients.dm:547]: did you forget to delet this

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Pull request merged**